# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/tschmidt/graphviz/gvconfig_hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./lib/gvc/gvconfig.c.orig	2005-08-18 11:25:03.000000000 +0200
+++ ./lib/gvc/gvconfig.c	2005-08-18 11:28:17.000000000 +0200
@@ -235,7 +235,12 @@
     gvplugin_library_t *library;
     char *plugin_glob = "libgvplugin*.so.?";
 
-    if (config_path) {
+    // this so-called "config file" should be in /var/cache, morrons!
+    // here is a little hack which skips updating the file when graphviz is
+    // called in a Build-Pkg process tree (unless when building graphviz).
+    // The variable ROCK_BUILDING_GRAPHVIZ must be set by the graphviz.conf
+    // script when building graphviz.  - clifford
+    if (config_path && (!getenv("ROCKCFG_ID") || getenv("ROCK_BUILDING_GRAPHVIZ"))) {
 	f = fopen(config_path,"w");
 	if (!f) {
 	    agerr(AGERR,"failed to open %s for write.\n", config_path);
